packages/purpose-process/src/services/purposeService.ts

```diff
@@ -734,6 +734,340 @@ export function purposeServiceBuilder(
+        fromState,
+      }: {
+        fromState:
+          | typeof purposeVersionState.draft
+          | typeof purposeVersionState.waitingForApproval;
+      }): Promise<{
+        event: CreateEvent<PurposeEvent>;
+        updatedPurposeVersion: PurposeVersion;
+      }> {
+        const updatedPurposeVersion: PurposeVersion = {
+          ...purposeVersion,
+          state: purposeVersionState.active,
+          riskAnalysis: await generateRiskAnalysisDocument({
+            eservice,
+            purpose: purpose.data,
+            dailyCalls: purposeVersion.dailyCalls,
+            readModelService,
+            storeFile: fileManager.storeBytes,
+            logger,
+          }),
+          updatedAt: new Date(),
+          firstActivationAt: new Date(),
+        };
+
+        const updatedPurpose: Purpose = replacePurposeVersion(
+          {
+            ...purpose.data,
+            versions: archiveActiveAndSuspendedPurposeVersions(
+              purpose.data.versions
+            ),
+          },
+          updatedPurposeVersion
+        );
+
+        if (fromState === purposeVersionState.draft) {
+          return {
+            event: toCreateEventPurposeActivated({
+              purpose: updatedPurpose,
+              version: purpose.metadata.version,
+              correlationId,
+            }),
+            updatedPurposeVersion,
+          };
+        } else {
+          return {
+            event: toCreateEventPurposeVersionActivated({
+              purpose: updatedPurpose,
+              versionId: updatedPurposeVersion.id,
+              version: purpose.metadata.version,
+              correlationId,
+            }),
+            updatedPurposeVersion,
+          };
+        }
+      }
+
+      function activateFromSuspended(): {
+        event: CreateEvent<PurposeEvent>;
+        updatedPurposeVersion: PurposeVersion;
+      } {
+        const newState = match({
+          suspendedByProducer: purpose.data.suspendedByProducer,
+          suspendedByConsumer: purpose.data.suspendedByConsumer,
+          ownership,
+        })
+          .with(
+            {
+              suspendedByConsumer: true,
+              ownership: "PRODUCER",
+            },
+            {
+              suspendedByProducer: true,
+              ownership: "CONSUMER",
+            },
+            () => purposeVersionState.suspended
+          )
+          .otherwise(() => purposeVersionState.active);
+
+        const updatedPurposeVersion: PurposeVersion = {
+          ...purposeVersion,
+          updatedAt: new Date(),
+          suspendedAt:
+            newState !== purposeVersionState.suspended
+              ? undefined
+              : purposeVersion.suspendedAt,
+          state: newState,
+        };
+
+        const updatedPurpose: Purpose = replacePurposeVersion(
+          purpose.data,
+          updatedPurposeVersion
+        );
+
+        if (ownership === "PRODUCER" || ownership === "SELF_CONSUMER") {
```

Comment:
You could replace the strings with `ownership.PRODUCER` and `ownership.SELF_CONSUMER`.